prog/weaveproxy/main.go

```diff
@@ -57,6 +59,10 @@ func main() {
 	Log.Infoln("weave proxy", version)
 	Log.Infoln("Command line arguments:", strings.Join(os.Args[1:], " "))
 
+	if withDNS {
+		Log.Warning("--with-dns option has been removed; DNS is on by default")
```

Comment:
in fact I'd make the message identical to that in the script, i.e. "--with-dns is deprecated; it is on by default".